# Информация о географическом расположении клиентов (geolocation_olist_public_dataset.csv)

In [1]:
import pandas as pd

In [2]:
geolocation_data = pd.read_csv(r'C:\Users\Вячеслав\Documents\olist_churn\olist-ml-churn-prediction\data\raw\geolocation_olist_public_dataset.csv')

In [10]:
display(geolocation_data.head())
print('Размерность:', geolocation_data.shape)
print(geolocation_data.dtypes)

zip_code_prefix       city state        lat        lng
0               10  sao paulo    sp -23.547807 -46.636122
1               10  sao paulo    sp -23.546081 -46.644820
2               10  sao paulo    sp -23.540984 -46.642569
3               10  sao paulo    sp -23.546758 -46.645771
4               10  sao paulo    sp -23.546362 -46.643073

Размерность: (323016, 5)
zip_code_prefix      int64
city                object
state               object
lat                float64
lng                float64
dtype: object


## Трактовка признаков:

#### *zip_code_prefix* - это префикс почтового индекса. До 3 цифр;

#### *city* - город заказчика;

#### *state* - штат;

#### *lat* - анг. latitude, широта. Координаты ненастоящие, созданы случайно на основе принадлежности к определенному zip_code_prefix;

#### *lng* - анг. longitude, долгота. Также ненастоящие.

In [11]:
print('Пропуски:', geolocation_data.isna().sum())

Пропуски: zip_code_prefix    0
city               0
state              0
lat                0
lng                0
dtype: int64


### Пропусков нет

In [5]:
print(geolocation_data['zip_code_prefix'].sort_values())

0          10
341        10
340        10
339        10
338        10
         ... 
322766    999
322765    999
322764    999
322762    999
323015    999
Name: zip_code_prefix, Length: 323016, dtype: int64


### Значения zip_code_prefix лежат в промежутке [10; 999]

print(geolocation_data['zip_code_prefix'].value_counts())

### Распределение по zip_code_prefix неравномерное, но потолок количества пользователей для одного zip_code_prefix - 500

In [18]:
print(len(geolocation_data['city'].unique()))
print(len(geolocation_data['state'].unique()))

5419
27


### 5419 уникальных городов, 27 уникальных штатов